x-pack/plugins/security/public/management/api_keys/api_keys_grid/create_api_key_flyout.tsx

```diff
@@ -0,0 +1,358 @@
+      submitButtonText={i18n.translate(
+        'xpack.security.accountManagement.createApiKey.submitButton',
+        {
+          defaultMessage: '{isSubmitting, select, true{Creating API key…} other{Create API key}}',
+          values: { isSubmitting: form.isSubmitting },
+        }
+      )}
+      isLoading={form.isSubmitting}
+      isDisabled={isLoading || (form.isSubmitted && form.isInvalid)}
+      size="s"
+      ownFocus
+    >
+      {form.submitError && (
+        <>
+          <EuiCallOut
+            title={i18n.translate('xpack.security.accountManagement.createApiKey.errorMessage', {
+              defaultMessage: 'Could not create API key',
+            })}
+            color="danger"
+          >
+            {(form.submitError as any).body?.message || form.submitError.message}
+          </EuiCallOut>
+          <EuiSpacer />
+        </>
+      )}
+
+      {isLoading ? (
+        <EuiLoadingContent />
+      ) : (
+        <EuiForm
+          component="form"
+          isInvalid={form.isInvalid}
+          error={Object.values(form.errors)}
+          invalidCallout={!form.submitError && form.isSubmitted ? 'above' : 'none'}
+          {...eventHandlers}
+        >
+          <EuiFormRow
+            label={i18n.translate(
+              'xpack.security.management.users.changePasswordFlyout.userLabel',
+              {
+                defaultMessage: 'User',
+              }
+            )}
+          >
+            <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false}>
+              <EuiFlexItem grow={false}>
+                <EuiIcon type="user" />
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiSpacer size="xs" />
+                <EuiText>{currentUser?.username}</EuiText>
+                <EuiSpacer size="xs" />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiFormRow>
+
+          <EuiFormRow
+            label={i18n.translate('xpack.security.accountManagement.createApiKey.nameLabel', {
+              defaultMessage: 'Name',
+            })}
+            helpText={i18n.translate('xpack.security.accountManagement.createApiKey.nameHelpText', {
+              defaultMessage: 'What is this key used for?',
+            })}
+            error={form.errors.name}
+            isInvalid={form.touched.name && !!form.errors.name}
+          >
+            <EuiFieldText
+              name="name"
+              defaultValue={form.values.name}
+              isInvalid={form.touched.name && !!form.errors.name}
+              inputRef={firstFieldRef}
+              fullWidth
+            />
+          </EuiFormRow>
+
+          <EuiSpacer />
+          <EuiFormFieldset>
+            <EuiSwitch
+              id="apiKeyCustom"
+              label={i18n.translate(
+                'xpack.security.accountManagement.createApiKey.customPrivilegesLabel',
+                {
+                  defaultMessage: 'Restrict access and permissions',
```

Comment:
I think we can simplify the language here and just call these "privileges"

```suggestion
                  defaultMessage: 'Restrict privileges',
```